src/status_im/protocol/handlers.cljs

```diff
@@ -1,291 +1,204 @@
-            :sent                  (let [{:keys [to id group-id]} message
-                                         message' {:from    to
-                                                   :payload {:message-id id
-                                                             :group-id   group-id}}]
-                                     (dispatch [:message-sent message']))
-            (debug "Unknown message type" type)))))))
+(re-frame/reg-fx
+  ::web3-get-syncing
+  (fn [web3]
+    (when web3
+      (.getSyncing
+        (.-eth web3)
+        (fn [error sync]
+          (re-frame/dispatch [:update-sync-state error sync]))))))
+
+(re-frame/reg-fx
+  ::save-processed-messages
+  (fn [processed-message]
+    (processed-messages/save processed-message)))
 
 (defn system-message
   ([message-id timestamp content]
    {:from         "system"
     :message-id   message-id
     :timestamp    timestamp
     :content      content
-    :content-type text-content-type}))
-
-(defn joined-chat-message [chat-id from message-id]
-  (let [contact-name (:name (contacts/get-by-id from))]
-    (messages/save chat-id {:from         "system"
-                            :message-id   (str message-id "_" from)
-                            :content      (str (or contact-name from) " " (label :t/received-invitation))
-                            :content-type text-content-type})))
-
-(defn participant-invited-to-group-message [chat-id current-identity identity from message-id timestamp]
-  (let [inviter-name (:name (contacts/get-by-id from))
-        invitee-name (if (= identity current-identity)
-                       (label :t/You)
-                       (:name (contacts/get-by-id identity)))]
-    {:from         "system"
-     :group-id     chat-id
-     :timestamp    timestamp
-     :message-id   message-id
-     :content      (str (or inviter-name from) " " (label :t/invited) " " (or invitee-name identity))
-     :content-type text-content-type}))
-
-(defn participant-removed-from-group-message
-  [identity from {:keys [message-id timestamp]}]
-  (let [remover-name (:name (contacts/get-by-id from))
-        removed-name (:name (contacts/get-by-id identity))]
-    (->> [(or remover-name from) (label :t/removed) (or removed-name identity)]
-         (join " ")
-         (system-message message-id timestamp))))
+    :content-type constants/text-content-type}))
+
+(re-frame/reg-fx
+  ::participant-removed-from-group-message
+  (fn [{:keys [identity from message-id timestamp group-id]}]
+    (let [remover-name (:name (contacts/get-by-id from))
+          removed-name (:name (contacts/get-by-id identity))
+          message (->> [(or remover-name from) (i18n/label :t/removed) (or removed-name identity)]
+                       (string/join " ")
+                       (system-message message-id timestamp))
+          message' (assoc message :group-id group-id)]
+      (re-frame/dispatch [:received-message message']))))
```

Comment:
Why is this even effect, all that it does is bunch of operations on app-data followed by dispatch.

Reply:
because it reads data from data-store, i want to get rid of it